.github/workflows/ci.yml

```diff
@@ -0,0 +1,75 @@
+name: CI
+
+on:
+  push:
+    branches: [ main]
+  pull_request:
+    branches: [ main ]
+
+env:
+  CARGO_TERM_COLOR: always
+  APPNAME : test-flip-link-app
+  CORE_TARGET: thumbv7em-none-eabi # needed by `core`
+  NRF_TARGET: thumbv7em-none-eabihf
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        rust:
+          - stable
+          #- nightly
+        os:
+          - ubuntu-latest
+          # FIXME: Reenable macOS, Windows, and nightly once demt-test is open-sourced.
+          #- macOS-latest
+          #- windows-latest
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 10
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: ${{ matrix.rust }}
+        override: true
+    - name: Build
+      run: RUSTFLAGS='--deny warnings' cargo build
+      shell: bash
+    - name: Check fmt
+      if: matrix.os == 'ubuntu-latest' # we only need to check fmt on one OS
+      run: cargo fmt --all -- --check
+
+  build_app:
+    strategy:
+      matrix:
+        rust:
+          - stable
+          #- nightly
+        os:
+          - ubuntu-latest # doesn't work on macOS because `cortex-m-rt` doesn't build there
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.rust }}
+          override: true
+          target: ${{ env.NRF_TARGET }}
+      - name: Install dependencies
+        run: |
+          sudo apt-get update && sudo apt-get install libssl-dev
+          rustup target add ${{ env.CORE_TARGET }}
+          cargo install cargo-generate
+          cargo install --path .
```

Comment:
we can cut down compilation here by not optimizing `flip-link`: pass `--debug` to `cargo install`. Runtime perf of `flip-link` is not too important because the bulk of the work is done by `rust-lld`.